ansible_runner/cleanup.py

```diff
@@ -0,0 +1,191 @@
+import datetime
+import glob
+import os
+import signal
+import subprocess
+import sys
+
+from pathlib import Path
+from tempfile import gettempdir
+
+from ansible_runner.defaults import GRACE_PERIOD_DEFAULT
+from ansible_runner.defaults import registry_auth_prefix
+from ansible_runner.utils import cleanup_folder
+
+
+__all__ = ['add_cleanup_args', 'run_cleanup']
+
+
+def add_cleanup_args(command):
+    command.add_argument(
+        "--file-pattern",
+        help="A file glob pattern to find private_data_dir folders to remove. "
+             "Example: --file-pattern=/tmp/.ansible-runner-*"
+    )
+    command.add_argument(
+        "--exclude-strings",
+        nargs='*',
+        help="A comma separated list of keywords in directory name or path to avoid deleting."
+    )
+    command.add_argument(
+        "--remove-images",
+        nargs='*',
+        help="A comma separated list of podman or docker tags to delete. "
+             "This may not remove the corresponding layers, use the image-prune option to assure full deletion. "
+             "Example: --remove-images=quay.io/user/image:devel quay.io/user/builder:latest"
+    )
+    command.add_argument(
+        "--grace-period",
+        default=GRACE_PERIOD_DEFAULT,
+        type=int,
+        help="Time (in minutes) after last modification to exclude a folder from deletion for. "
+             "This is to avoid deleting folders that were recently created, or folders not started via the start command. "
+             "Value of 0 indicates that no folders will be excluded based on modified time."
+    )
+    command.add_argument(
+        "--image-prune",
+        action="store_true",
+        help="If specified, will run docker / podman image prune --force. "
+             "This will only run after untagging."
+    )
+    command.add_argument(
+        "--process-isolation-executable",
+        default="podman",
+        help="The container image to clean up images for (default=podman)"
+    )
+
+
+def run_command(cmd):
+    '''Given list cmd, runs command and returns standard out, expecting success'''
+    process = subprocess.run(cmd, capture_output=True)
+    stdout = str(process.stdout, encoding='utf-8')
+    if process.returncode != 0:
+        print('Error running command:')
+        print(' '.join(cmd))
+        print('Stdout:')
+        print(stdout)
+        raise RuntimeError('Error running command')
+    return stdout.strip()
+
+
+def is_alive(dir):
+    pidfile = os.path.join(dir, 'pid')
+
+    try:
+        with open(pidfile, 'r') as f:
+            pid = int(f.readline())
+    except IOError:
+        return False
+
+    try:
+        os.kill(pid, signal.SIG_DFL)
```

Comment:
Why the choice of `SIG_DFL` here? According to the [signal docs](https://docs.python.org/3/library/signal.html#signal.SIG_DFL) this would usually cause a core dump and exit. The usual pattern is to first use `SIGTERM` and then maybe `SIGKILL` if you need to be more forceful.

Reply:
Okay, what I _should_ have done is to refactor existing code here:

https://github.com/ansible/ansible-runner/blob/4f67a3a9e2f20e5d4da313958881eeef67397011/ansible_runner/__main__.py#L914

Into a `is_alive` method that lives somewhere else. However terrible the choice of this logic is, I do not have any intention of addressing it with this change set.

Reply:
Ah, you simply moved the code. Yes, I agree you shouldn't need to address that here. Thx for the explanation.